src/ORM/DataObject.php

```diff
@@ -3724,6 +3724,40 @@ public function onAfterBuild()
         $this->extend('onAfterBuild');
     }
 
+    private function hasRelations($fieldPath)
+    {
+        $component = $this;
+        foreach (explode('.', $fieldPath ?? '') as $relation) {
+            if (!$component) {
+                return null;
+            }
+            $hasRelation = function () use ($component, $relation) {
+                if ($component instanceof DataObject) {
```

Comment:
If this is false, we don't return any value... should this whole function just be this instead?
```php
return ($component instanceof DataObject) && $component->getRelationType($relation) !== null;
```

Reply:
Updated